cylc/flow/workflow_files.py

```diff
@@ -1698,6 +1698,24 @@ def get_run_dir_info(
     return relink, run_num, rundir
 
 
+def detect_both_flow_and_suite(path):
+    """Detects if both suite.rc and flow.cylc are in directory.
+    Raises:
+        WorkflowFilesError: If both flow.cylc and suite.rc are in directory
+    """
+    msg = (f"Both {WorkflowFiles.FLOW_FILE} and {WorkflowFiles.SUITE_RC} "
+           "files are present in the run directory. Please remove one and"
+           " try again.")
+    if path.resolve().name == WorkflowFiles.SUITE_RC:
+        flow_cylc = path.parent / WorkflowFiles.FLOW_FILE
+        if flow_cylc.is_file() and not flow_cylc.is_symlink():
+            raise WorkflowFilesError(msg)
+    elif (path / WorkflowFiles.SUITE_RC).is_file():
+        flow_cylc = path / WorkflowFiles.FLOW_FILE
+        if flow_cylc.is_file() and not flow_cylc.is_symlink():
+            raise WorkflowFilesError(msg)
```

Comment:
```suggestion
    if path.resolve().name == WorkflowFiles.SUITE_RC:
        flow_cylc = path.parent / WorkflowFiles.FLOW_FILE
    elif (path / WorkflowFiles.SUITE_RC).is_file():
        flow_cylc = path / WorkflowFiles.FLOW_FILE

    if flow_cylc.is_file() and not flow_cylc.is_symlink():
        raise WorkflowFilesError(msg)
```

Reply:
Discussed on phone: This doesn't work because flow_cylc is only defined inside the first if..else loop.